data/theater.1
.\" Copyright (C) 2004 Andre Lehovich <andrel@u.arizona.edu>
.\"
.\" This is free software; you may redistribute it and/or modify
.\" it under the terms of the GNU General Public License as
.\" published by the Free Software Foundation; either version 2,
.\" or (at your option) any later version.
.\"
.\" This is distributed in the hope that it will be useful, but
.\" WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License 
.\" along with this; if not write to the Free Software Foundation, Inc.
.\" 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
.TH theater 1 "2008\-08\-25" "GNOME"
.SH NAME
theater \- GNOME desktop movie player based on GStreamer
.SH SYNOPSIS
.B theater
.RI [ option... ] " " [ filenames | URI ]
.SH DESCRIPTION
.B theater
is a movie player for the GNOME desktop.  Through use of the
.B GStreamer
backend many popular file and streaming formats are supported including:
MPEG audio and video (MP3/MPV), Ogg, AVI, ASF, MOV, WMV,
QuickTime, and more.  Playing of DVD and VCD media is
also possible.
Although primarily intended as a movie player,
.B theater
can also play many audio formats.
.P
For additional documentation on interactive use of 
.B theater
select the
.I Help
menu option.  The menus allow access to some features not
available through keybindings or command-line options.
.SH OPTIONS
.TP
.B \-?, --help
Output a longer help message then quit.  (Any discrepancies
between this man page and the output of
.B theater --help
should be reported as a documentation bug.)
.TP
.B --version
Output version information then quit.
.P
The following options command an already-running instance of
.B theater
to do something; they are useful for remote-control scripting. If 
.B theater
is not already running, these commands will launch a new instance of
.B theater
but will not do anything further.
.TP
.B --play-pause
Tell an already-running instance of 
.B theater
to toggle between play and pause.
.TP
.B --play
Tell an already-running instance of
.B theater
to play (has no effect if already playing)
.TP
.B --pause
Tell an already-running instance of
.B theater
to pause (has no effect if already paused)
.TP
.B --next
Tell an already-running instance of 
.B theater
to skip to the next movie or chapter in the playlist.
.TP
.B --previous
Tell an already-running instance of 
.B theater
to return to the previous movie or chapter in the playlist.
.TP
.B --seek-fwd
Tell an already-running instance of 
.B theater
to seek forwards 15 seconds in the current movie.
.TP
.B --seek-bwd
Tell an already-running instance of 
.B theater
to seek backwards 15 seconds in the current movie.
.TP
.B --volume-up
Tell an already-running instance of 
.B theater
to raise the volume by 8%.
.TP
.B --volume-down
Tell an already-running instance of 
.B theater
to lower the volume by 8%.
.TP
.B --mute
Tell an already-running instance of 
.B theater
to toggle between muted and unmuted audio.
.TP
.B --fullscreen
Tell an already-running instance of 
.B theater
to toggle fullscreen mode.
.TP
.B --quit
Tell an already-running instance of
.B theater
to quit.
.TP
.BI "--enqueue " filename|URI
Tell an already-running instance of theater to add a new stream
to the playlist.
.TP
.BI "--replace " filename|URI
Tell an already-running instance of theater to play 
from the playlist.
.SH KEYBINDINGS
Global keybindings for
.BR theater :
.P
.ta \w'Down-arrow   'u	
a	cycle between aspect ratios
.br
p	toggle between play and pause
.br
Esc	exit full screen mode
.br
f	toggle full screen
.br
h	toggle display of on-screen controls
.br
0	resize window to 50% original size
.br
1	resize window to 100% original size
.br
2	resize window to 200% original size
.br
.br
r	zoom in the video
.br
t	zoom out the video
.br
d	start and stop the telestrator (drawing) mode
.br
e	erase the drawing
.br
Left-arrow        skip back 15 seconds
.br
Right-arrow       skip forward 60 seconds
.br
Shift+Left-arrow  skip back 5 seconds
.br
Shift+Right-arrow skip forward 15 seconds
.br
Ctrl+Left-arrow   skip back 3 minutes
.br
Ctrl+Right arrow  skip forward 10 minutes
.br
Up-arrow	increase volume by 8%
.br
Down-arrow	decrease volume by 8%
.br
b		jump back to previous chapter/movie in playlist
.br
n		jump to next chapter/movie in playlist
.br
q		quit
.br
Ctrl+E          eject the playing optical media
.br
Ctrl+O		open a new file
.br
Ctrl+L		open a new URI
.br
F9		toggle display of the playlist
.br
m		show the DVD menu
.br
c		show the DVD chapter menu
.SH FILES
.I /usr/lib/theater/plugins/
.RS
The global plugins directory, which contains all the system-installed
plugins.
.RE
.I ~/.local/share/theater/plugins/
.RS
The user plugins directory, containing plugins available only to that user.
It is always in the "theater/plugins" subdirectory of your 
.B XDG_DATA_HOME
directory.
.RE
.I ~/.config/theater/state.ini
.RS
The configuration file for the main window's appearance, stored in a
simple INI format.
.RE
.SH BUGS
.B theater
cannot play files using some proprietary codecs.
.SH AUTHOR
.B theater
was written by Bastien Nocera <hadess@hadess.net>.
This manual page was written by Andre Lehovich for the
Debian Project.
.SH DISTRIBUTION
The latest version of theater may be downloaded from
.UR https://wiki.gnome.org/Apps/Videos
<https://wiki.gnome.org/Apps/Videos>
.UE
.SH SEE ALSO
.BR "theater-video-thumbnailer" (1),
.P
The online documentation available through the program's
.I Help
menu.


NEWS
New features and significant updates in version...

Major changes in 3.34.1:
- Fix broken symbolic icon
- Translation updates

Major changes in 3.34.0:
- Make introspection non-optional for the plugin system to work
- Translation updates

Major changes in 3.33.90:
- Update application icon
- Use title case for the "Open Containing Folder" menu item
- Disable VAAPI support for now, it's too buggy

Major changes in 3.33.0:
- Add "Open containing folder" menu item
- Include "theater" in the desktop keywords
- Make --version command-line option public
- Reduce GSettings writes on startup
- Fix menu sensitivity for OpenSubtitles plugin
- Remove obsolete LIRC and Brasero plugins

Major changes in 3.32.1:
- Show the application window when opening a new file
- Work-around "Can't play a text file without video or visualizations" error
- Build fixes
- Translation updates and fixes

Major changes in 3.32.0:
- Translation updates

Major changes in 3.31.92:
- Add an error message when file is encrypted, or empty,
  or DVD playback is attempted with missing libraries
- Make "Stop" keyboard button act like the Stop button on a
  CD player
- Remove all mentions of the external chapters plugin, it wasn't
  used since 2012
- Fix possible crash when browsing
- Fix DesktopEntry in MPRIS plugin
- Translation updates

Major changes in 3.31.91:
- Move nautilus properties page to nautilus module
- Fix fullscreen toolbar's appearance in RTL languages
- Update appdata and screenshots
- Don't advertise VCD support, GStreamer 1.0 has never supported it
- Add support for "start time" in playlists
- Make Ctrl+W navigate back before exiting
- Make it possible to load subtitles from the command-line
- Fix media player controls not appearing in gnome-shell in
  some cases, and the next/previous buttons not working
- Make previous/next buttons' state update when a movie
  contains chapters
- Use better labels for Subtitles and Languages menus
- Fix some display problems when files are longer than detected
- And other smaller bug fixes
  - Throw a warning when download buffering breaks because
    of a missing plugin
  - Make framerate more precise in properties
  - Fix a warning on exit when brasero is not installed

Major changes in 3.31.90:
- New icon
- Remove app menu
- Modernise user documentation
- Loads of fixes in "Make available offline" functionality
- Fix proxy configuration in Flatpak
- Disable VA-API plugins in thumbnailer
- Plenty of build fixes and changes, including:
  - Use gettext instead of intltool
  - Fix parallel build
  - Remove gromit, ontop and zeitgeist plugins
  - Remove most X11 dependencies (except XInitThreads)

Major changes in 3.26.1:
- Fix crash when properties tab is closed too fast
- Thumbnail OPUS and M4B audio files
- Fix thumbnailing on platforms with V4L M2M hardware decoders
- Fix playing back MJPEG files
- Fix Python Console menu not appearing
- Build fixes:
  - Remove dependency on Debian-specific pycompile utility
  - Fix build with gtk-doc 1.26

Major changes in 3.26.0:
- Fix a possible build race
- Fix lirc plugin compilation
- Python console bug fixes
- Install appdata in the new location
- Translation updates

Major changes in 3.25.90.1:
- Build fixes

Major changes in 3.25.90:
- Move to meson build system
- Fix media-player keys usage to match API documentation
- Fix internal video widget API documentation
- Remove unused theater-audio-preview
- Split off video thumbnailer into thumbnailer and gallery maker
- Blacklist bmcdec plugin in the thumbnailer

Major changes in 3.24.0:
- Translation updates
- Add Flatpak build definition

Major changes in 3.23.90:
- Add keyboard shortcut to de-select all items
- Fix possible crash on first run
- Set a minimum width for the "Go" menu
- Mark many audio and video mime-types as handled
- Add more shortcuts to the keyboard shortcuts dialogue
- Fix a small memory leak in thumbnailer
- Fix missing plugins' startup notification on Wayland
- Fix crash when network turns off
- Port to non-obsoleted widgets
- and build fixes

Major changes in 3.22.0:
- Fix symbolic icon not appearing in gnome-shell's top bar

Major changes in 3.21.91:
- Add variable rate playback menu
- Build fixes
- Fix thumbnailing of 4k videos

Major changes in 3.20.1:
- Make playback rate available to plugins
- Add more shortcuts for the shortcuts help window
- Fix a number of compilation warnings

Major changes in 3.20.0:
- Fix a theming regression in the play/pause button box
- Translation updates

Major changes in 3.19.92:
- Fix regression from 3.18 which could cause none of the builtin
  plugins to be activated

Major changes in 3.19.91:
- Change "Skip to" label in shortcuts
- Make header popup translucent in fullscreen as per mockups
- Fix theater-audio-preview translations
- Add context to strings to translate in the shortcut window

Major changes in 3.19.3:
- Port to grilo 0.3.0
- Add a shortcuts help window
- Use new grl-video-title-parsing plugin
- Move controls OSD to the bottom of the screen
- Fix fullscreen toolbar not appearing in Wayland
- Add Ctrl+W to quit
- Add man page for theater-audio-preview
- Fix GSettings changing on startup
- A number of deprecation warning fixes

Major changes in 3.18.1:
- Make DVD menu navigation work again
- Updated translations

Major changes in 3.18.0:
- Fix rendering bugs in the search entry

Major changes in 3.17.92:
- None compared to 3.16.4

Major changes in 3.16.4:
- Fix selection not working after visiting Channels page
- Fix MPRIS plugin following internal API changes
- Revert Wayland compatibility changes, it's not a required
  change in GNOME 3.16

Major changes in 3.16.3:
- Fix ASF videos not being thumbnailed
- Bring back video inside the window under Wayland
- Fix Skip To dialogue not working
- Fix thumbnailer with newer versions of gstreamer-vaapi installed
- Update YouTube API key

Major changes in 3.16.2:
- Fix popup not showing up on touchscreens
- Fix potential crash on 32-bit systems

Major changes in 3.16.1:

src/theater.c
/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*- */
/* 
 * Copyright (C) 2001-2007 Bastien Nocera <hadess@hadess.net>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA.
 *
 * The theater project hereby grant permission for non-gpl compatible GStreamer
 * plugins to be used and distributed together with GStreamer and theater. This
 * permission are above and beyond the permissions granted by the GPL license
 * theater is covered by.
 *
 * Monday 7th February 2005: Christian Schaller: Add exception clause.
 * See license_change file for details.
 *
 */

#include "config.h"

#include <glib.h>
#include <glib-object.h>
#include <glib/gi18n.h>
#include <gtk/gtk.h>
#include <locale.h>
#include <string.h>

#ifdef GDK_WINDOWING_X11
/* X11 headers */
#include <X11/Xlib.h>
#endif

#include "theater.h"
#include "theater-private.h"
#include "theater-interface.h"
#include "theater-options.h"
#include "theater-menu.h"
#include "theater-uri.h"
#include "theater-preferences.h"
#include "theater-session.h"

int
main (int argc, char **argv)
{
	theater *theater;

	setlocale (LC_ALL, "");
	bindtextdomain (GETTEXT_PACKAGE, GNOMELOCALEDIR);
	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
	textdomain (GETTEXT_PACKAGE);

#ifdef GDK_WINDOWING_X11
	if (XInitThreads () == 0)
	{
		gtk_init (&argc, &argv);
		g_set_application_name (_("Videos"));
		theater_object_show_error_and_exit (_("Could not initialize the thread-safe libraries."), _("Verify your system installation. theater will now exit."), NULL);
	}
#endif

	g_set_prgname ("theater");
	g_set_application_name (_("Videos"));
	gtk_window_set_default_icon_name ("org.gnome.theater");
	g_setenv("PULSE_PROP_media.role", "video", TRUE);
	g_setenv("PULSE_PROP_application.icon_name", "org.gnome.theater", TRUE);

	/* Build the main theater object */
	theater = g_object_new (theater_TYPE_OBJECT,
			      "application-id", "org.gnome.theater",
			      "flags", G_APPLICATION_HANDLES_OPEN,
			      NULL);

	g_application_run (G_APPLICATION (theater), argc, argv);

	return 0;
}
